snapcraft/projects.py

```diff
@@ -64,6 +64,71 @@ def _validate_command_chain(command_chains: Optional[List[str]]) -> Optional[Lis
     return command_chains
 
 
+def _expand_architectures(architectures):
+    """Expand architecture data.
+
+    Expansion to fully-defined Architecture objects includes the following:
+        - strings (shortform notation) are converted to Architecture objects
+        - `build-on` and `build-to` strings are converted to single item sets
+        - Empty `build-to` fields are implicitly set to the same architecture used in `build-on`
+    """
+    for index, architecture in enumerate(architectures):
+        # convert strings into Architecture objects
+        if isinstance(architecture, str):
```

Comment:
this should no longer be valid, it confuses a lot of people, we are in the right place to remove support for a list of scalars

Reply:
as agreed earlier we should now leave this

Reply:
:facepalm:  I ripped this out but improved the scalar to list conversion for `build-to` and `build-on`.

Reply:
Restored.